node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -230,11 +231,34 @@ case class DataMessageHandler(
           this.copy(chainApi = newApi, syncing = newSyncing)
         }
       case msg: BlockMessage =>
+        val block = msg.block
         logger.info(
-          s"Received block message with hash ${msg.block.blockHeader.hash.flip}")
-        callbacks
-          .executeOnBlockReceivedCallbacks(logger, msg.block)
-          .map(_ => this)
+          s"Received block message with hash ${block.blockHeader.hash.flip}")
+
+        val newApiF = {
+          // if we've already processed this header it will fail, but we should still try
```

Comment:
Why don't we just check if we have it in our DB? I guess this doesn't _necessarily_ mean that `.processHeader()` completed sucessfully but it seems better than this no? 

The problem with the current implementation is postgres does not throw `SQLiteException` :-)

Reply:
That's probably a better idea